packages/transaction-controller/src/utils/gas-fees.ts

```diff
@@ -0,0 +1,279 @@
+    return initialParams.gasPrice;
+  }
+
+  if (suggestedGasFees.gasPrice) {
+    log('Using suggested gasPrice', suggestedGasFees.gasPrice);
+    return suggestedGasFees.gasPrice;
+  }
+
+  log('gasPrice not set');
+  return undefined;
+}
+
+function getUserFeeLevel(request: GetGasFeeRequest): UserFeeLevel | undefined {
+  const { eip1559, initialParams, suggestedGasFees, txMeta } = request;
+
+  if (!eip1559) {
+    return undefined;
+  }
+
+  if (
+    !initialParams.maxFeePerGas &&
+    !initialParams.maxPriorityFeePerGas &&
+    initialParams.gasPrice
+  ) {
+    return txMeta.origin === ORIGIN_METAMASK
+      ? UserFeeLevel.CUSTOM
+      : UserFeeLevel.DAPP_SUGGESTED;
+  }
+
+  if (
+    !initialParams.maxFeePerGas &&
+    !initialParams.maxPriorityFeePerGas &&
+    suggestedGasFees.maxFeePerGas &&
+    suggestedGasFees.maxPriorityFeePerGas
+  ) {
+    return UserFeeLevel.MEDIUM;
+  }
+
+  if (txMeta.origin === ORIGIN_METAMASK) {
+    return UserFeeLevel.MEDIUM;
+  }
+
+  return UserFeeLevel.DAPP_SUGGESTED;
+}
+
+function updateDefaultGasEstimates(txMeta: TransactionMeta) {
+  if (!txMeta.defaultGasEstimates) {
+    txMeta.defaultGasEstimates = {};
+  }
+
+  txMeta.defaultGasEstimates.maxFeePerGas = txMeta.txParams.maxFeePerGas;
+
+  txMeta.defaultGasEstimates.maxPriorityFeePerGas =
+    txMeta.txParams.maxPriorityFeePerGas;
+
+  txMeta.defaultGasEstimates.gasPrice = txMeta.txParams.gasPrice;
+  txMeta.defaultGasEstimates.estimateType = txMeta.userFeeLevel;
+}
+
+async function getSuggestedGasFees(request: UpdateGasFeesRequest) {
+  const { eip1559, ethQuery, getGasFeeEstimates, txMeta } = request;
+
+  if (
+    (!eip1559 && txMeta.txParams.gasPrice) ||
+    (eip1559 &&
+      txMeta.txParams.maxFeePerGas &&
+      txMeta.txParams.maxPriorityFeePerGas)
+  ) {
+    return {};
+  }
+
+  try {
+    const { gasFeeEstimates, gasEstimateType } = await getGasFeeEstimates();
+
+    if (eip1559 && gasEstimateType === GAS_ESTIMATE_TYPES.FEE_MARKET) {
+      const {
+        medium: { suggestedMaxPriorityFeePerGas, suggestedMaxFeePerGas } = {},
+      } = gasFeeEstimates;
+
+      if (suggestedMaxPriorityFeePerGas && suggestedMaxFeePerGas) {
+        return {
+          maxFeePerGas: gweiDecimalToWeiHex(suggestedMaxFeePerGas),
+          maxPriorityFeePerGas: gweiDecimalToWeiHex(
+            suggestedMaxPriorityFeePerGas,
+          ),
+        };
+      }
+    }
+
+    if (gasEstimateType === GAS_ESTIMATE_TYPES.LEGACY) {
+      // The LEGACY type includes low, medium and high estimates of
+      // gas price values.
+      return {
+        gasPrice: gweiDecimalToWeiHex(gasFeeEstimates.medium),
+      };
+    }
+
+    if (gasEstimateType === GAS_ESTIMATE_TYPES.ETH_GASPRICE) {
+      // The ETH_GASPRICE type just includes a single gas price property,
+      // which we can assume was retrieved from eth_gasPrice
+      return {
+        gasPrice: gweiDecimalToWeiHex(gasFeeEstimates.gasPrice),
+      };
+    }
+  } catch (e) {
+    console.error(e);
```

Comment:
Is there a specific reason for not utilizing the `log` function here?

Reply:
Done.